packages/core/src/app/customer/StripeGuestForm.tsx

```diff
@@ -69,10 +70,13 @@ const StripeGuestForm: FunctionComponent<StripeGuestFormProps & FormikProps<Gues
     }, [setContinueAsAGuestButton, onChangeEmail]);
 
     useEffect(() => {
+        if (status?.valSuccess) {
```

Comment:
Where is this variable set? Is this set on stripe's side?

Reply:
_Status_ if provided by _withFormik_ Hoc. It is set by _setStatus_ in _handleSubmit_ at the bottom of file where _StripeGuestForm_ is wrapped by _withFormik_.

Reply:
Thanks, I think we should split this effect into it's own block rather than wrap it in already used one.

Reply:
Something like

```
useEffect(() => {
        if (status?.valSuccess) {
             handleOnClickSubmitButton()
        }
}, [status?.valSuccess])
```